### ISSUE TITLE
dbus hide/show operation disables tray

### ISSUE THREAD

**Issue by kpbochenek** [travitch/taffybar#219]
Hi! (@IvanMalison)

When I do hide taffybar then show I have tray icons gone.

Also when taffybar runs on screen "0" and try to activate it on screen "1" I have only tray icons on screen "0".

When I run taffybar only on screen "0" and do 'hide' then 'show' icons are gone and only restart helps.

Cheers,
Krzysiek

**Comment by IvanMalison** [comment]
Yeah, this is a known (sort of) issue. The problem is that only 1 widget can own any given tray widget at a time. You'll notice that if you start an application like trayer before taffybar, the tray will also fail to show up, but of course what is actually happening is that all of the tray icons are simply not registering with the taffybar tray because they've already registered with trayer.

I think the correct course of action would be to explicitly destroy and unregister the tray when taffybar is "hidden" on a given monitor, but when I tried to add a haskell binding to the c function that does this for the tray it started causing crashes.

I've also tried making it so that there is actually only 1 tray-manager widget and it is simply moved between the different taffybars. This also failed for some reason.
